kytos/core/user.py

```diff
@@ -42,38 +72,15 @@ def create_salt(cls, salt):
 class UserDoc(DocumentBaseModel):
     """UserDocumentModel."""
 
-    username: constr(min_length=1, max_length=64, regex="^[a-zA-Z0-9_-]+$")
+    username: str = Field(
+        min_length=1, max_length=64, pattern=r'^[a-zA-Z0-9_-]+$'
+    )
     hash: HashSubDoc
     state: Literal['active', 'inactive'] = 'active'
     email: EmailStr
-    password: constr(min_length=8, max_length=64)
-
-    @validator('password')
-    def validate_password(cls, password, values):
-        """Check if password has at least a letter and a number"""
-        upper = False
-        lower = False
-        number = False
-        for char in password:
-            if char.isupper():
-                upper = True
-            if char.isnumeric():
-                number = True
-            if char.islower():
-                lower = True
-            if number and upper and lower:
-                return cls.hashing(password.encode(), values['hash'].dict())
-        raise ValueError('value should contain ' +
-                         'minimun 8 characters, ' +
-                         'at least one upper case character, ' +
-                         'at least 1 numeric character [0-9]')
+    password: str = Field(min_length=8, max_length=64)
 
-    @staticmethod
-    def hashing(password: bytes, values: dict) -> str:
-        """Hash password and return it as string"""
-        return hashlib.scrypt(password=password, salt=values['salt'],
-                              n=values['n'], r=values['r'],
-                              p=values['p']).hex()
+    _validate_password = field_validator('password')(validate_password)
```

Comment:
`pydantic` had lots of changes. This includes the changed from `@validator` to `@field_validator`. With the new implementation I had to move `validate_password` outside of the model class.